src/bin/genemichaels.rs

```diff
@@ -250,14 +351,149 @@ fn main() {
+        );
         if failed {
             process::exit(1);
         }
     }
 }
+
+fn process_cargo_toml(path: PathBuf, manifest: Manifest, config: FormatConfig) {
+    let mut dirs = manifest.workspace.map(|wkspc| wkspc.members).map(|mb| {
+        let mut rust_dirs: Vec<PathBuf> =
+            mb.into_iter().filter_map(|m| path.join(&m).exists().then(|| path.join(m))).collect();
+        if let Some(bins) = manifest.bin {
+            for bin in bins {
+                rust_dirs.push(
+                    match bin
+                        .path
+                        .and_then(|p| p.parse::<PathBuf>().ok())
+                        .and_then(|p| p.parent().map(|pp| pp.to_path_buf())) {
+                        Some(p) => p,
+                        None => {
+                            // default location
+                            if path.join("bin").exists() {
+                                rust_dirs.push(path.join("bin"));
+                            }
+                            continue;
+                        },
+                    },
+                )
+            }
+        }
+        if let Some(lib) = manifest.lib {
+            if let Some(p) =
+                lib
+                    .path
+                    .and_then(|p| p.parse::<PathBuf>().ok())
+                    .and_then(|p| p.parent().map(|pp| pp.to_path_buf())) {
+                rust_dirs.push(p)
+            }
+        }
+        if let Some(benches) = manifest.bench {
+            for bench in benches {
+                rust_dirs.push(
+                    match bench
+                        .path
+                        .and_then(|p| p.parse::<PathBuf>().ok())
+                        .and_then(|p| p.parent().map(|pp| pp.to_path_buf())) {
+                        Some(p) => p,
+                        None => {
+                            // default location
+                            if path.join("benches").exists() {
+                                rust_dirs.push(path.join("benches"));
+                            }
+                            continue;
+                        },
+                    },
+                )
+            }
+        }
+        if let Some(tests) = manifest.test {
+            for test in tests {
+                rust_dirs.push(
+                    match test
+                        .path
+                        .and_then(|p| p.parse::<PathBuf>().ok())
+                        .and_then(|p| p.parent().map(|pp| pp.to_path_buf())) {
+                        Some(p) => p,
+                        None => {
+                            // default location
+                            if path.join("tests").exists() {
+                                rust_dirs.push(path.join("tests"));
+                            }
+                            continue;
+                        },
+                    },
+                )
+            }
+        }
+        if let Some(examples) = manifest.example {
```

Comment:
With this Cargo.toml
```
[package]
name = "x103"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]

[workspace]

[[example]]
name = "hi"
path = "exampleb/hi.rs"
```
and another example in `examples/hello.rs` it formatted `hi.rs` but not `hello.rs`.

Reply:
Yeah ok, thats a weird one, unexpected tbh - realistically what this means is the default paths should be checked for existence regardless of the toml.

Reply:
Yeah, I think that's right (as far as my experimentation goes). Basically there's no way to change the directories, just add extras outside of the main directories.

Reply:
[See above](https://github.com/andrewbaxter/genemichaels/pull/9#discussion_r1060209896)